### PR TITLE
Use fastify

### PR DESCRIPTION
**Pull request by mcollina**
I have a few more changes to make here, so I'm going to flag it as a draft.

### CODE REVIEW

File: exercises/05.actions/05.solution.history-revalidation/server/app.js [hunk 1]
```diff
@@ -35,31 +55,32 @@ app.use((req, res, next) => {
 		next()
 	}
 })
+*/
```
Comment (mcollina) [inline exercises/05.actions/05.solution.history-revalidation/server/app.js]: @kentcdodds I really don't understand this part.

File: exercises/05.actions/05.solution.history-revalidation/server/app.js [hunk 2]
```diff
@@ -35,31 +55,32 @@ app.use((req, res, next) => {
...
 }
 
-app.get('/rsc/:shipId?', async (req, res) => {
-	await renderApp(res, null)
+app.get('/rsc/:shipId?', (req, res) => {
+	res.type('text/html').send(renderApp(res, null))
 })
 
-app.post('/action/:shipId?', bodyParser.text(), async (req, res) => {
+app.post('/action/:shipId?', async (req, res) => {
```
Comment (mcollina) [inline exercises/05.actions/05.solution.history-revalidation/server/app.js:69]: How do I trigger this action from the frontend? I couldn't find it.

Comment (kentcdodds) [reply exercises/05.actions/05.solution.history-revalidation/server/app.js:69]: It's tricky. You click on the name of a ship and it changes to an edit form.

File: exercises/05.actions/05.solution.history-revalidation/server/app.js (hunk 1 above)
Comment (kentcdodds) [reply exercises/05.actions/05.solution.history-revalidation/server/app.js]: The url can be: `/someId?search=` and I want that to change to `/someId` just to keep it clean.

**Comment by mcollina** [comment]
I think it's all working. Let me know if there is anything not working.

File: exercises/05.actions/05.solution.history-revalidation/server/app.js [hunk 3]
```diff
@@ -1,66 +1,73 @@
...
-import busboy from 'busboy'
 import closeWithGrace from 'close-with-grace'
-import compress from 'compression'
-import express from 'express'
+import fastify from 'fastify'
+import compress from '@fastify/compress'
+import fastifyStatic from '@fastify/static'
+import multipart from '@fastify/multipart'
 import { createElement as h } from 'react'
+import { PassThrough } from 'stream'
```
Comment (kibertoad) [inline exercises/05.actions/05.solution.history-revalidation/server/app.js:7]: why not node
```suggestion
import { PassThrough } from 'node:stream'
```

File: exercises/05.actions/05.solution.history-revalidation/server/app.js [hunk 4]
```diff
@@ -1,66 +1,73 @@
...
 import {
 	renderToPipeableStream,
-	decodeReplyFromBusboy,
+	decodeReply
 } from 'react-server-dom-esm/server'
 import { App } from '../src/app.js'
 import { shipDataStorage } from './async-storage.js'
+import { join } from 'path'
```
Comment (kibertoad) [inline exercises/05.actions/05.solution.history-revalidation/server/app.js:14]:
```suggestion
import { join } from 'node:path'
```

File: exercises/05.actions/05.solution.history-revalidation/server/app.js [hunk 5]
```diff
@@ -1,66 +1,73 @@
...
+    const payload = { root, returnValue }
+
+    const { pipe } = renderToPipeableStream(payload, moduleBasePath)
+    return pipe(new PassThrough())
+  })
 }
 
-app.get('/rsc/:shipId?', async (req, res) => {
-	await renderApp(res, null)
+app.get('/rsc/:shipId?', (req, res) => {
```
Comment (kibertoad) [inline exercises/05.actions/05.solution.history-revalidation/server/app.js:65]: is question mark needed here?

**Comment by kentcdodds** [comment]
This is working great! The biggest thing that's drawing me towards hono is the web standard APIs of Request/Response. Not having to manually change the request body into a FormData object for example (I like that hono's request allows me to get the formData directly).

**Comment by mcollina** [comment]
If the only blocker is adding a `req.formdata()`... that doesn't seem hard to add. It's actually a great addition.

**Comment by kentcdodds** [comment]
I think that would be a good addition, but it's not the only blocker. It's just generally the benefit of using web standard objects. It means there's much less documentation I need to look up. I can just use request/response as I use them elsewhere. Nothing special.

**Comment by mcollina** [comment]
`Response` is fully supported, so if you have one you can return it and it would work:

https://fastify.dev/docs/latest/Reference/Reply/#response

(the return value from the handler is passed to `reply.send()`)

---

`req.formdata()`: https://github.com/fastify/fastify-multipart/pull/522

**Review by o-az — APPROVED** [review]

**Comment by o-az** [comment]
Sorry I was looking at this through GitHub mobile app and somehow approved the PR. I don't know how to remove it.

**Comment by kentcdodds** [comment]
Thanks for this. I've decided to go with Hono.js.

**Pull request closed without merging**